Riot/Modules/Room/RoomViewController.m

```diff
@@ -3474,13 +3476,15 @@ - (void)roomTitleView:(RoomTitleView*)titleView recognizeTapGesture:(UITapGestur
                     // If an event was specified, replace the datasource by a non live datasource showing the event
                     if (eventId)
                     {
-                        RoomDataSource *roomDataSource = [[RoomDataSource alloc] initWithRoomId:self.roomDataSource.roomId initialEventId:eventId andMatrixSession:self.mainSession];
-                        [roomDataSource finalizeInitialization];
-                        roomDataSource.markTimelineInitialEvent = YES;
-                        
-                        [self displayRoom:roomDataSource];
-                        
-                        self.hasRoomDataSourceOwnership = YES;
+                        [RoomDataSource loadRoomDataSourceWithRoomId:self.roomDataSource.roomId initialEventId:eventId andMatrixSession:self.mainSession onComplete:^(id roomDataSource) {
```

Comment:
We can add weak self reference in block to not retain self if process take a lot of time